src/test/groovy/netflix/nebula/dependency/recommender/DependencyRecommendationsPluginMultiprojectCoreBomSupportSpec.groovy

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016-2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package netflix.nebula.dependency.recommender
+
+import nebula.test.IntegrationSpec
+import nebula.test.dependencies.DependencyGraphBuilder
+import nebula.test.dependencies.GradleDependencyGenerator
+import nebula.test.dependencies.ModuleBuilder
+import nebula.test.dependencies.maven.ArtifactType
+import nebula.test.dependencies.maven.Pom
+import nebula.test.dependencies.repositories.MavenRepo
+
+class DependencyRecommendationsPluginMultiprojectCoreBomSupportSpec extends IntegrationSpec {
+    def repo
+    def generator
+
+    def setup() {
+        fork = true
+        new File("${projectDir}/gradle.properties").text = "systemProp.nebula.features.coreBomSupport=true"
```

Comment:
The implementation has `Boolean.getBoolean("nebula.features.coreBomSupport")` as the way to find out if core bom support is enabled. Since this is a property, how is this handled across a multiproject project? Does the root properties file take precedence or the nearest property...?

Reply:
It's a java system property instead of a project property, so there will only be one per JVM. A convenient way to set that is in the root project's gradle.properties (which is the only one that will be evaluated):

> In a multi project build, “systemProp.” properties set in any project except the root will be ignored. That is, only the root project’s gradle.properties file will be checked for properties that begin with the “systemProp.” prefix.

This is the [order of precedence](https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties):

1. gradle.properties in project root directory.
2. gradle.properties in GRADLE_USER_HOME directory.
3. system properties, e.g. when -Dgradle.user.home is set on the command line.

Reply:
Thanks!